packages/grid/_modules_/grid/components/GridRow.tsx

```diff
@@ -133,6 +142,41 @@ function GridRow(props: React.HTMLAttributes<HTMLDivElement> & GridRowProps) {
     [apiRef, rowId],
   );
 
+  const publishClick = React.useCallback(
+    (event) => {
+      const cell = findParentElementFromClassName(event.target, gridClasses.cell);
+      const field = cell?.getAttribute('data-field');
+
+      // Check if the field is available because the cell that fills the empty
+      // space of the row has no field.
+      if (field) {
+        // User clicked in the checkbox added by checkboxSelection
+        if (field === GRID_CHECKBOX_SELECTION_COL_DEF.field) {
+          return;
+        }
+
+        // User opened a detail panel
+        if (field === GRID_DETAIL_PANEL_TOGGLE_FIELD) {
+          return;
+        }
+
+        // User is editing a cell
+        if (apiRef.current.getCellMode(rowId, field) === GridCellModes.Edit) {
+          return;
+        }
```

Comment:
Except if some users complained about it, I'm wondering if is not better to wait for v6 for this case. In v6 we could change the behavior, such that a row is not selected when clicking on a cell in edit mode.

This would allow keeping row selection sync with `handleRowClick`

Reply:
> In v6 we could change the behavior, such that a row is not selected when clicking on a cell in edit mode.

To be clear, this PR changed to listen to `cellClick` to select rows. To prevent selection on click we have the `disableSelectionOnClick` prop which is enabled in most demos that use cell editing.

The change you highlighted I did it only to keep in sync with the idea that we don't fire `rowClick` when clicking on cells with interactive elements. In reality, binding something to the row click (like in #3805 where it opens a detail panel) + cell editing might have accessibility implications because the click outside the current cell (which can be in another cell) is used to cancel the editing.